scripts/lint-style.py

```diff
@@ -295,10 +303,16 @@ def format_errors(errors):
             output_message(path, line_nr, "ERR_DOT", "Line is an isolated focusing dot or uses . instead of ·")
         if errno == ERR_SEM:
             output_message(path, line_nr, "ERR_SEM", "Line contains a space before a semicolon")
+        if errno == ERR_WIN:
+            output_message(path, line_nr, "ERR_WIN", "Windows line endings (\\r\\n) detected")
 
 def lint(path):
-    with path.open(encoding="utf-8") as f:
+    with path.open("rb") as f:
```

Comment:
```suggestion
    with path.open(encoding="utf-8", newline="") as f:
```
seems like a simpler way to do this